homeassistant/components/sql/sensor.py

```diff
@@ -58,67 +51,101 @@ def validate_sql_select(value: str) -> str:
 )
 
 
-def setup_platform(
+async def async_setup_platform(
     hass: HomeAssistant,
     config: ConfigType,
-    add_entities: AddEntitiesCallback,
+    async_add_entities: AddEntitiesCallback,
     discovery_info: DiscoveryInfoType | None = None,
 ) -> None:
     """Set up the SQL sensor platform."""
-    if not (db_url := config.get(CONF_DB_URL)):
-        db_url = DEFAULT_URL.format(hass_config_path=hass.config.path(DEFAULT_DB_FILE))
+    _LOGGER.warning(
+        # SQL config flow added in 2022.4 and should be removed in 2022.6
+        "Configuration of the SQL sensor platform in YAML is deprecated and "
+        "will be removed in Home Assistant 2022.6; Your existing configuration "
+        "has been imported into the UI automatically and can be safely removed "
+        "from your configuration.yaml file"
+    )
+
+    default_db_url = DEFAULT_URL.format(
+        hass_config_path=hass.config.path(DEFAULT_DB_FILE)
+    )
 
-    sess: scoped_session | None = None
-    try:
-        engine = sqlalchemy.create_engine(db_url)
-        sessmaker = scoped_session(sessionmaker(bind=engine))
-
-        # Run a dummy query just to test the db_url
-        sess = sessmaker()
-        sess.execute("SELECT 1;")
-
-    except sqlalchemy.exc.SQLAlchemyError as err:
-        _LOGGER.error(
-            "Couldn't connect using %s DB_URL: %s",
-            redact_credentials(db_url),
-            redact_credentials(str(err)),
+    for query in config[CONF_QUERIES]:
+        new_config = {
+            CONF_DB_URL: config.get(CONF_DB_URL, default_db_url),
+            CONF_NAME: query.get(CONF_NAME),
+            CONF_QUERY: query.get(CONF_QUERY),
```

Comment:
Name and query are required items in the schema so we should not use `dict.get(key)` but `dict[key]`.